src/app/InteractionModelEngine.h

```diff
@@ -58,7 +58,7 @@ namespace chip {
 namespace app {
 
 constexpr size_t kMaxSecureSduLengthBytes = 1024;
-constexpr uint32_t kImMessageTimeoutMsec  = 3000;
+constexpr uint32_t kImMessageTimeoutMsec  = 6000;
```

Comment:
I presume that if this change is here, it is needed.  But what's the rationale?  It is clear to me how we manage CRMP timeouts.  But what about timeouts at this layer?  I suppose they have to exist in the case of unreliable exchanges.  But do we have any guidance for this in the spec?  What about interactions over CRMP?  Shouldn't timeouts at this layer be derived from CRMP-layer timeouts?

Reply:
The 3000 (3 seconds) timeout was causing OpCSR request to timeout. The device generates a CSR, that involves crypto operations. So it might take more time than other IM interactions. At 3 seconds it was very flaky (where it was mostly timing out).

I agree that the timeout should be more in tune with CRMP timeouts and retries. If it is less than CRMP retransmit timer (i.e. less than `number of retries` * `retransmit timeout`) it won't give CRMP a chance to retry dropped packets. Not sure if Specifications have any suggested value for it.

Reply:
Maybe then add a TODO?